gulpfile.ts

```diff
@@ -36,52 +45,124 @@ async function fetchContentsByPath(repoPath = 'models'): Promise<GithubObject[]>
+      if (error) {
+        reject(error)
+      } else {
+        resolve(model)
+      }
+    })
+  })
 }
 
-function removeRedundantObjects(model: APIModel): APIModel {
+async function removeRedundantObjects(model: APIModel): Promise<APIModel> {
   /**
-   * TODO: clean up:
+   * Clean up:
    *- .openapi-generator
    *- .gitignore
    *- .openapi-generator-ignore
    *- git_push.sh
    */
 
+  await Promise.all([
+    ...REDUNDANT_FILES.map((object) => fsPromises.unlink(`${model.outputPath}/${object}`)),
+    ...REDUNDANT_DIRECTORIES.map((object) =>
+      fsPromises.rmdir(`${model.outputPath}/${object}`, { recursive: true }),
+    ),
+  ])
+
   return model
 }
 
-function exportAPIModel(model: APIModel): APIModel {
-  // TODO: export models into src/index.ts and commit files
+/**
+ * Verify Data Models (Schemas) in OpenAPI definitions.
+ * OpenAPI definitions have various data models. Such as: array, boolean, object.
+ * We don't need to export all of them. Only export: concrete objects and enums.
+ *
+ * @param definitions Record<string, any>
+ * @param key string
+ * @returns boolean
+ */
+function verifyObjectDefinition(definitions: Record<string, any>, key: string): boolean {
+  const definitionType = definitions[key].type
+
+  return (
+    (definitionType === 'object' &&
+      !EXCLUDE_EXPORTED_OBJECTS.has(key) &&
+      /**
+       * The 'properties' keyword is used to define the object properties.
+       * Docs: https://swagger.io/docs/specification/data-models/data-types/#ctxM:~:text=The%20properties%20keyword%20is%20used%20to%20define%20the%20object%20properties
+       */
+      has(definitions[key], 'properties') &&
+      /**
+       * 'additionalProperties' is used to define a free form object.
+       * Docs: https://swagger.io/docs/specification/data-models/dictionaries/#free-form:~:text=Free%2DForm%20Objects
+       */
+
+      isEmpty(definitions[key].additionalProperties)) ||
+    /**
+     * Use to combine schemas
+     * Docs: https://swagger.io/docs/specification/data-models/oneof-anyof-allof-not/
+     */
+    has(definitions[key], 'oneOf') ||
+    has(definitions[key], 'allOf') ||
+    has(definitions[key], 'oneOf') ||
+    /**
+     *
+     */
+    (definitionType === 'string' && has(definitions[key], 'enum'))
+  )
+}
 
-  return model
+async function generateExportStatement(model: APIModel): Promise<string> {
+  return SwaggerParser.parse(model.download_url).then(({ definitions }) => {
+    const exportings: string[] = []
+
+    for (const key in definitions) {
+      if (has(definitions, key) && verifyObjectDefinition(definitions, key)) {
+        exportings.push(`${key} as ${upperFirst(camelCase(model.modelName))}${key}`)
+      }
+    }
+
+    return `export { ${exportings.join(', ')} } from './${model.modelName}'`
+  })
+}
+
+function writeStatementsToFile(statements: string[]): void {
+  return fs.writeFileSync('src/api-models/index.ts', statements.join('\n'))
```

Comment:
Why is this sync when everything else is async?

Reply:
@moltar : I don't want to await an async function while I can make it to sync. 
And performance isn't also optimized when I use async in there. 😕